sfm_pc/management/commands/download_data.py

```diff
@@ -0,0 +1,222 @@
+        )
+
+    def handle(self, *args, **kwargs):
+        entity_doc_id = kwargs['entity_doc_id']
+        location_doc_id = kwargs['location_doc_id']
+        sources_doc_id = kwargs['sources_doc_id']
+        country_code = kwargs['country_code'].rstrip()
+        
+        parent_directory = f'data'
+        country_directory = f'{parent_directory}/countries/{country_code}'
+
+        self._create_csv_files(
+            entity_doc_id=entity_doc_id,
+            sources_doc_id=sources_doc_id,
+            country_directory=country_directory,
+            sources_directory=parent_directory
+        )
+
+        self._create_location_file(location_doc_id, country_directory)
+    
+    def _create_location_file(self, location_doc_id, output_directory):
+        downloader, buffer = self._build_location_downloader(location_doc_id)
+        
+        self.stdout.write('Downloading locations file...')
+        
+        done = False
+
+        with tqdm(total=100) as progress_bar:
+            while not done:
+                status, done = downloader.next_chunk()
+                progress_bar.update(int(status.progress() * 100) - progress_bar.n)
+
+        buffer.seek(0)
+        
+        target = f'{output_directory}/locations.geojson'
+
+        with open(target, 'wb') as f:
+            f.write(buffer.getbuffer())
+
+        self.stdout.write(
+            self.style.SUCCESS('Downloaded locations file to {}'.format(target))
+        )
+    
+    def _build_location_downloader(self, location_doc_id):
+        drive_service = self._build_google_service(
+            scopes=['https://www.googleapis.com/auth/drive.readonly'],
+            service='drive',
+            version='v3'
+        )
+
+        request = drive_service.files().get_media(fileId=location_doc_id)
+        
+        location_buffer = io.BytesIO()
+        return MediaIoBaseDownload(location_buffer, request), location_buffer
+    
+    def _create_csv_files(
+            self,
+            entity_doc_id,
+            sources_doc_id,
+            country_directory,
+            sources_directory
+        ):
+        
+        sheets_service = self._build_google_service(
+            scopes=['https://www.googleapis.com/auth/spreadsheets.readonly'],
+            service='sheets',
+            version='v4'
+        )
+        
+        self._create_sources_file(
+            sheets_service=sheets_service,
+            document_id=sources_doc_id,
+            output_directory=sources_directory
+        )
+        
+        self._create_entity_files(
+            sheets_service=sheets_service,
+            document_id=entity_doc_id,
+            output_directory=country_directory
+        )
+
+    def _create_sources_file(self, sheets_service, document_id, output_directory):
+        target = f'{output_directory}/sources.csv'
+        
+        # No need to keep downloading the same data
+        if os.path.exists(target):
+            return
```

Comment:
if i re-combine the two csv functions with the key lambda, and call this method, then it will skip in the case of a fresh import. so we need to be sure to add a way in make to remove these files after the import finishes.